src/ProxyClient/AbstractProxyClient.php

```diff
@@ -120,196 +69,58 @@ protected function getBaseUrl()
-    {
-        return $this->client->createRequest($method, $url, $headers);
-    }
-
-    /**
-     * Sends all requests to each caching proxy server
-     *
-     * Requests are sent in parallel to minimise impact on performance.
-     *
-     * @param RequestInterface[] $requests Requests
-     *
-     * @throws ExceptionCollection
-     */
-    private function sendRequests(array $requests)
-    {
-        $allRequests = array();
-
-        foreach ($requests as $request) {
-            $headers = $request->getHeaders()->toArray();
-            // Force to re-create Host header if empty, as Apache chokes on this. See #128 for discussion.
-            if (empty($headers['Host'])) {
-                unset($headers['Host']);
-            }
-            foreach ($this->servers as $server) {
-                $proxyRequest = $this->createRequest(
-                    $request->getMethod(),
-                    $server.$request->getResource(),
-                    $headers
-                );
-                $allRequests[] = $proxyRequest;
-            }
-        }
+        $queue = clone $this->queue;
+        $this->queue->clear();
 
         try {
-            $this->client->send($allRequests);
-        } catch (GuzzleExceptionCollection $e) {
-            $this->handleException($e);
-        }
-    }
-
-    /**
-     * Handle request exception
-     *
-     * @param GuzzleExceptionCollection $exceptions
-     *
-     * @throws ExceptionCollection
-     */
-    protected function handleException(GuzzleExceptionCollection $exceptions)
-    {
-        $collection = new ExceptionCollection();
-
-        foreach ($exceptions as $exception) {
-            if ($exception instanceof CurlException) {
-                // Caching proxy unreachable
-                $e = ProxyUnreachableException::proxyUnreachable(
-                    $exception->getRequest()->getHost(),
-                    $exception->getMessage(),
-                    $exception->getRequest()->getRawHeaders(),
-                    $exception
-                );
-            } elseif ($exception instanceof RequestException) {
-                // Other error
-                $e = ProxyResponseException::proxyResponse(
-                    $exception->getRequest()->getHost(),
-                    $exception->getCode(),
-                    $exception->getMessage(),
-                    $exception->getRequest()->getRawHeaders(),
-                    $exception
-                );
-            } else {
-                // Unexpected exception type
-                $e = $exception;
+            $this->httpAdapter->sendRequests($queue->all());
+        } catch (MultiHttpAdapterException $e) {
+            $collection = new ExceptionCollection();
+            foreach ($e->getExceptions() as $exception) {
+                // A workaround for php-http currently lacking differentiation
+                // between client, server and networking errors.
```

Comment:
Actually no, that option cannot be overridden. That would allow adapters to behave differently.

I am not quite sure how exceptions should be handled. Currently no exception is thrown if a response is available.